### DISCUSSION TITLE
Script tags in _document.js only load properly on page refresh

### DISCUSSION THREAD

**Discussion by nicolauria** [vercel/next.js#22178] [Help]
I have a very basic nextjs app where I am trying to add html, css and js. I have created a '_document.js' file to load my script tags. It looks something like this:
```
import Document, { Html, Head, Main, NextScript } from 'next/document'

class MyDocument extends Document {
  static async getInitialProps(ctx) {
    const initialProps = await Document.getInitialProps(ctx)
    return { ...initialProps }
  }

  render() {
    return (
      <Html>
        <Head />
        <body>
          <Main />
          <NextScript />
          <script type="text/javascript" src="js/jquery-3.3.1.js"></script>
          <script type="text/javascript" src="js/jquery-migrate-3.0.0.min.js"></script>
          <script type="text/javascript" src="js/popper.min.js"></script>
          <script type="text/javascript" src="js/bootstrap.min.js"></script>
          <script type="text/javascript" src="js/jquery.mmenu.all.js"></script>
          <script type="text/javascript" src="js/ace-responsive-menu.js"></script>
          <script type="text/javascript" src="js/bootstrap-select.min.js"></script>
          <script type="text/javascript" src="js/isotop.js"></script>
          <script type="text/javascript" src="js/snackbar.min.js"></script>
          <script type="text/javascript" src="js/simplebar.js"></script>
          <script type="text/javascript" src="js/parallax.js"></script>
          <script type="text/javascript" src="js/scrollto.js"></script>
          <script type="text/javascript" src="js/jquery-scrolltofixed-min.js"></script>
          <script type="text/javascript" src="js/jquery.counterup.js"></script>
          <script type="text/javascript" src="js/wow.min.js"></script>
          <script type="text/javascript" src="js/slider.js"></script>
          <script type="text/javascript" src="js/pricing-slider.js"></script>
          <script type="text/javascript" src="js/timepicker.js"></script>
          <script type="text/javascript" src="js/script.js"></script>
        </body>
      </Html>
    )
  }
}

export default MyDocument
```
I have only created two pages for the time being. index.js and about.js

My problem is that the link to the about page never resolves page loading. I am using 'next/link'. The page spinner does not stop unless the page is refreshed.  

Here is the repo: https://github.com/nicolauria/next-balkon
Here is the deployed application: https://next-balkon.vercel.app/

Any help is greatly appreciated!

**Comment by michrome** [comment]
Hey @nicolauria it looks like [`initBalkon()`](https://github.com/nicolauria/next-balkon/blob/2246ae370feea9fbdf14cfdfb8219829725a760c/public/js/scripts.js#L4) is where you're fading out your loader but you're only calling that function on [`$(document).ready`](https://github.com/nicolauria/next-balkon/blob/2246ae370feea9fbdf14cfdfb8219829725a760c/public/js/scripts.js#L554), which won't be triggered if you're using `next/link`.

**Reply by nicolauria** [reply]
Thanks @michrome. I tried moving the `initBalkon()` function call outside the `$(document).ready` but I get the same result.

**Reply by michrome** [reply]
I can see you [moved it out of `$(document).ready`](https://github.com/nicolauria/next-balkon/commit/819eaff5d31f11115d4bd0e45adc659074fd768f#diff-2aefe940a94941f84727a0224816fadbb73b5d64d9b05ad6415165ddd781158f) but what is triggering it?

Unlike an `<a>` tag, `next/link` isn't reloading the page: it's updating the contents of the page. That means `$(document).ready` doesn't happen and `public/js/scripts.js` isn't re-evaluated.

Given that `next/link` will prefetch by default, maybe you don't need a loading animation; or maybe an `<a>` tag would suit your needs better in this case?

**Reply by nicolauria** [reply]
Using an `<a>` tag without `Link` seems to work. I had no idea I could use `<a>` tags in such a manner. Thanks for your help @michrome!
